### ISSUE TITLE
The getAtMostNode method in the RM should throw a runtimeException if scriptResult contains an exception

### ISSUE THREAD

**Issue by activeeon-bot** [ow2-proactive/scheduling#231]
<a href="https://jira.activeeon.com/browse/SCHEDULING-237" title="SCHEDULING-237">Original issue</a> created by <a href="mailto:jean-luc.scheefer_AT_activeeon.com">Jean-Luc Scheefer</a> on 02, Apr 2009 at 12:22 PM - SCHEDULING-237

<hr />

Currently, it is not possible to know if a selection script is invalid or not when using the getAtMostNode() method.

It should be nice to have a runtimeException (let's call it 'SelectionScriptException') which is thrown by the getAtMostNode() method if the selectionScript is considered as invalid (can be syntax error, selected value not set, ...).
A invalid selection script would be a selection script that cannot answer yes. (We need the Yes to win against the no)

This issue has to be manage in the RMCore, maybe in the processScriptResults() method. Be careful to free busy nodes if this code has been executed (RMCore:932):
if (scriptPassed) {
                try {
                    internalSetBusy(rmnode);
                    result.add(rmnode.getNode());
                } catch (NodeException e) {
                    internalSetDown(rmnode);
                }
            }

To sum up, if a selection script result contain an exception, getAtMostNode must return a runtimeException and the alredy busied node should be released.
